aea/mail/base.py

```diff
@@ -50,13 +50,12 @@ def __init__(self, to: Address,
         self._sender = sender
         self._protocol_id = protocol_id
         self._message = message
-        assert type(self._to) == str or self._to is None
-        try:
+        try : 
+            assert type(self._to) == str or self._to is None
             if self._to is not None and type(self._to) == str:
                 self._to.encode('utf-8')
-        except Exception:
-            assert False
-
+        except AssertionError as error : 
```

Comment:
the handling of the failing assertion is a good idea. However, we should decide if we tolerate this bad state or not. if we raise another exception or we just log a warning/error/exception message.